### PR TITLE
Fix compiling and running in Linux-64bit with fpc 3.2.0

### PR DESCRIPTION
**Pull request by cutec-chris**
Sry, i have not written enougth in the last Pull Request.
The ifdefś in the windows units make sense for Lazarus, becouse in Lazarus you dont use an Global Search Path but Packages. And those Packages are used across plattforms. So the Windows Units are included in the Linux Packages also (like in the Synapse Lazarus Package). I hope the 4 ifdefs dont disturb someone.

The second fix is the RTTI Handling in fpc 3.2.0 and up seems to be has changed in last time additionally https://wiki.freepascal.org/User_Changes_Trunk#Type_information_contains_reference_to_attribute_table So FPC_PROVIDE_ATTR_TABLE should be enabled for fpc 3.2.0 and 3.3.1 also.

### CODE REVIEW

File: SynSSPI.pas [hunk 1]
```diff
@@ -55,7 +55,7 @@
 {$I Synopse.inc} // define HASINLINE and other compatibility switches
 
 interface
-
+{$ifdef WIONDOWS}
```
Comment (synopse) [inline SynSSPI.pas]: WINDOWS :)

File: Synopse.inc [hunk 2]
```diff
@@ -242,11 +242,13 @@
     {$define ISFPC32}
   {$endif}
   {$ifdef VER3_2}
+    {$define FPC_PROVIDE_ATTR_TABLE}
```
Comment (synopse) [inline Synopse.inc]: this is not correct, and would break compilation on 3.2/3.3

**Comment by synopse** [comment]
I guess the proper implementation is https://github.com/synopse/mORMot/commit/19986e9638b8d14363302c0dd294342350025a6d

File: Synopse.inc (hunk 2 above)
Comment (cutec-chris) [reply Synopse.inc]: have you tried it ?
i have it with 3.2 in use and the actual implementation dosend work...
can we please discuss this before you close the pull requests ?
or do you dont want that somebody help you ?

File: SynSSPI.pas (hunk 1 above)
Comment (cutec-chris) [reply SynSSPI.pas]: OK, it seems the same implementation as mine :)

**Comment by synopse** [comment]
Le 28 déc. 2019 08:24, "Christian U." <notifications@github.com> a écrit :@cutec-chris commented on this pull request.



In Synopse.inc:
> @@ -242,11 +242,13 @@
     {$define ISFPC32}
   {$endif}
   {$ifdef VER3_2}
+    {$define FPC_PROVIDE_ATTR_TABLE}

have you tried it ?
i have it with 3.2 in use and the actual implementation dosend work...
can we please discuss this before you close the pull requests ?
or do you dont want that somebody help you ?

—You are receiving this because you modified the open/close state.Reply to this email directly, view it on GitHub, or unsubscribe.
Le 28 déc. 2019 08:24, "Christian U." <notifications@github.com> a écrit :@cutec-chris commented on this pull request.



In Synopse.inc:
> @@ -242,11 +242,13 @@
     {$define ISFPC32}
   {$endif}
   {$ifdef VER3_2}
+    {$define FPC_PROVIDE_ATTR_TABLE}

have you tried it ?
i have it with 3.2 in use and the actual implementation dosend work...
can we please discuss this before you close the pull requests ?
or do you dont want that somebody help you ?

—You are receiving this because you modified the open/close state.Reply to this email directly, view it on GitHub, or unsubscribe.
In the trunk SVN revision Le 28 déc. 2019 08:24, "Christian U." <notifications@github.com> a écrit :@cutec-chris commented on this pull request.



In Synopse.inc:
> @@ -242,11 +242,13 @@
     {$define ISFPC32}
   {$endif}
   {$ifdef VER3_2}
+    {$define FPC_PROVIDE_ATTR_TABLE}

have you tried it ?
i have it with 3.2 in use and the actual implementation dosend work...
can we please discuss this before you close the pull requests ?
or do you dont want that somebody help you ?

—You are receiving this because you modified the open/close state.Reply to this email directly, view it on GitHub, or unsubscribe.
Le 28 déc. 2019 08:24, "Christian U." <notifications@github.com> a écrit :@cutec-chris commented on this pull request.



In Synopse.inc:
> @@ -242,11 +242,13 @@
     {$define ISFPC32}
   {$endif}
   {$ifdef VER3_2}
+    {$define FPC_PROVIDE_ATTR_TABLE}

have you tried it ?
i have it with 3.2 in use and the actual implementation dosend work...
can we please discuss this before you close the pull requests ?
or do you dont want that somebody help you ?

—You are receiving this because you modified the open/close state.Reply to this email directly, view it on GitHub, or unsubscribe.
In the trunk SVN revision in the documentation, this shouldn't be defined.We don't support latest trunk by default since it is very unstable.Le 28 déc. 2019 08:24, "Christian U." <notifications@github.com> a écrit :@cutec-chris commented on this pull request.



In Synopse.inc:
> @@ -242,11 +242,13 @@
     {$define ISFPC32}
   {$endif}
   {$ifdef VER3_2}
+    {$define FPC_PROVIDE_ATTR_TABLE}

have you tried it ?
i have it with 3.2 in use and the actual implementation dosend work...
can we please discuss this before you close the pull requests ?
or do you dont want that somebody help you ?

**Pull request closed without merging**